source/common/umf_helpers.hpp

```diff
@@ -279,6 +280,39 @@ inline ur_result_t umf2urResult(umf_result_t umfResult) {
   };
 }
 
+inline umf_result_t createUmfCUDAprovider(
+    umf_cuda_memory_provider_params_handle_t cu_memory_provider_params,
+    int cuDevice, void *cuContext, umf_usm_memory_type_t memType,
+    umf_memory_provider_handle_t *provider) {
+  umf_result_t umf_result = umfCUDAMemoryProviderParamsSetContext(
+      cu_memory_provider_params, cuContext);
+  if (umf_result != UMF_RESULT_SUCCESS) {
+    return umf_result;
+  }
```

Comment:
[NIT] I think we could wrap all if condition of this kind in a new macro that works like `UR_CHECK_RESULT`, or we could change this function to return `ur_result_t` and wrap this in `UR_CHECK_RESULT(umf::umf2urResult(umf_result))` so that makes the code more concise. I don't have strong opinions which one of this 2 approaches is better. but I think this function is used in UR entry points anyway, so the second approach might make its usage easier like instead of `UR_CHECK_ERROR(umf::umf2urResult(createUmfCUDAprovider(...)))`, it becomes `UR_CHECK_ERROR(createUmfCUDAprovider(...))`

Reply:
Done.